Added TLS for new units + multiple AC's

kevinbaptist opened the pull request:
I have acquired Gree Pulsar recently and they are using TLS to communicate with the server, so adding support for those.
+
I have multiple ACs so added support for multiple devices

kevinbaptist committed:
Added TLS for new units
Added multithread support when multiple AC's

spuke113 commented:
I tested this with Innova AC, but I get TLS handshake error: SSLError: [SSL: NO_SHARED_CIPHER] no shared cipher (_ssl.c:1000)

Wireshark capture shows the AC is trying handshake with TLS 1.2 but I guess the pyopenssl library supports only newer tls 1.3? I tried multiple ways to force TLS 1.2 but so far I failed. Any ideas?

kevinbaptist committed:
# Set cipher suites to include a broader range

kevinbaptist commented:
@spuke113 I've modified python script, potentially fixing your issue. In sum I've added 
# Set cipher suites to include a broader range
server_context.set_ciphers("ALL:@SECLEVEL=1")

markv9401 commented:
Wow, sorry for the late reply, I didn't notice only just now. Thanks for the fixes. I'll be accepting PRs and such but this project is end of life on my side as I moved on to converting the whole WiFi module to ESPHome. If you can, you probably should too. 

https://github.com/markv9401/esphome_external_components/tree/main/components/gree

(Based on multiple projects. Been working fine for months now)
